File: src/Other/MaterialDesign_Styles_Kit/TextToPlaceholderTextVisibilityConverter.cs
using System;
#if SLMIGRATION
using System.Windows;
using System.Windows.Data;
using System.Globalization;
#else
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;
#endif

namespace MaterialDesign_Styles_Kit
{
    public class TextToPlaceholderTextVisibilityConverter : IValueConverter
    {
#if SLMIGRATION
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
#else
        public object Convert(object value, Type targetType, object parameter, string language)
#endif
        {
            if (value is string)
            {
                if (string.IsNullOrEmpty((string)value))
                    return Visibility.Visible;
                return Visibility.Collapsed;
            }
            throw new InvalidOperationException("The IValueConverter expects a value of type String.");
        }

#if SLMIGRATION
        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
#else
        public object ConvertBack(object value, Type targetType, object parameter, string language)
#endif
        {
            throw new NotImplementedException();
        }
    }
}

File: src/Other/MaterialDesign_Styles_Kit/TextToPlaceholderTextVisibilityConverter.fs
﻿namespace MaterialDesign_Styles_Kit

open System
open System.Windows
open System.Windows.Data
open System.Globalization

type TextToPlaceholderTextVisibilityConverter() =
    interface IValueConverter with
        member this.Convert(value, targetType, parameter, language) =
            match value with
            | :? string as str ->
                if String.IsNullOrEmpty(str) then Visibility.Visible
                else Visibility.Collapsed
            | _ -> raise <| InvalidOperationException("The IValueConverter expects a value of type String.")

        member this.ConvertBack(value, targetType, parameter, language) =
            raise <| NotImplementedException()
